### DISCUSSION TITLE
Behaviour changed with matchDepTypes and rangeStrategy

### DISCUSSION THREAD

**Discussion by dubzzz** [renovatebot/renovate#23291] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

**Initial case**

On the project [fast-check](https://github.com/dubzzz/fast-check/), Renovate started to open PRs not matching the constraints declared on the configuration file starting on the 10th of July (roughly).

[My configuration](https://github.com/dubzzz/fast-check/blob/main/.github/renovate.json) comes with the following rules for packages:

```json
  "packageRules": [
    {
      "matchPaths": ["**/package.json"],
      "matchDepTypes": ["dependencies", "peerDependencies", "optionalDependencies"],
      "rangeStrategy": "update-lockfile"
    },
    {
      "matchPaths": ["**/package.json"],
      "matchDepTypes": ["devDependencies"],
      "rangeStrategy": "bump"
    },
    {
      "matchPaths": ["**/!(package.json)"],
      "rangeStrategy": "bump"
    }
  ],
```

In the past, Renovate used to increase the version of devDeps in both package.json and yarn.lock but only bumped others in the lock file. Now, Renovate also bumps versions linked to dependencies in the package.json itself (eg.: [this PR](https://github.com/dubzzz/fast-check/pull/4053)).

---

**Minimal repro**

A minimal repro is available at https://github.com/dubzzz/renovate-update-bug.

In that repro, I don't expect the PR touching fast-check bump to increase the version in the package.json but only to touch the lock file. For pure-rand I expect both to be touched.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
`matchPaths` was changed to `matchFileNames` in v36. It's meant to migrate with backwards compatibility but maybe that's not the case. Could you test with `matchFileNames` and see if there's any change in behavior? You should see a migration message in your logs too

**Reply by dubzzz** [reply]
I did the change but it does not seem to be enough. The opened PR still tries to bump in the package.json file (here https://github.com/dubzzz/renovate-update-bug/pull/1/files).

**Reply by rarkins** [reply, marked as answer]
Here's the problem: `matchFileNames` matches against both the "package file" (`package.json`) and the "lock file" (`package-lock.json`). So `**/!(package.json)` matches against `package-lock.json` even though you anticipate it won't match against `package.json`.

I recommend you use _positive_ matches instead of a negated one like that, even if it means one extra package rule.

**Reply by dubzzz** [reply]
Thanks a lot for your answer, it fixed my issue. The problem was indeed the lock file being matched by the negative glob
